module/documents/item.mjs

```diff
@@ -642,101 +642,169 @@ export default class Item5e extends Item {
-    let consumeUsage = !!uses.per;              // Consume limited uses
-    let consumeQuantity = uses.autoDestroy;     // Consume quantity of the item in lieu of uses
-    let consumeSpellLevel = null;               // Consume a specific category of spell slot
-    if ( requireSpellSlot ) consumeSpellLevel = is.preparation.mode === "pact" ? "pact" : `spell${is.level}`;
+    config = foundry.utils.mergeObject({
+      createMeasuredTemplate: item.hasAreaTarget,
+      consumeQuantity: is.uses?.autoDestroy ?? false,
+      consumeRecharge: !!is.recharge?.value,
+      consumeResource: !!resource.target && (!item.hasAttack || (resource.type !== "ammo")),
+      consumeSpellLevel: requireSpellSlot ? is.preparation.mode === "pact" ? "pact" : is.level : null,
+      consumeSpellSlot: requireSpellSlot,
+      consumeUsage: !!is.uses?.per
+    }, config);
 
     // Display a configuration dialog to customize the usage
-    const needsConfiguration = createMeasuredTemplate || consumeRecharge || consumeResource
-      || consumeSpellSlot || consumeUsage;
-    if (configureDialog && needsConfiguration) {
-      const configuration = await AbilityUseDialog.create(this);
-      if (!configuration) return;
-
-      // Determine consumption preferences
-      createMeasuredTemplate = Boolean(configuration.placeTemplate);
-      consumeUsage = Boolean(configuration.consumeUse);
-      consumeRecharge = Boolean(configuration.consumeRecharge);
-      consumeResource = Boolean(configuration.consumeResource);
-      consumeSpellSlot = Boolean(configuration.consumeSlot);
-
-      // Handle spell upcasting
-      if ( requireSpellSlot ) {
-        consumeSpellLevel = configuration.level === "pact" ? "pact" : `spell${configuration.level}`;
-        if ( consumeSpellSlot === false ) consumeSpellLevel = null;
-        const upcastLevel = configuration.level === "pact" ? as.spells.pact.level : parseInt(configuration.level);
-        if ( !Number.isNaN(upcastLevel) && (upcastLevel !== is.level) ) {
-          item = this.clone({"system.level": upcastLevel}, {keepId: true});
-          item.prepareData();
-        }
+    if ( config.needsConfiguration === undefined ) config.needsConfiguration = config.createMeasuredTemplate
+      || config.consumeRecharge || config.consumeResource || config.consumeSpellSlot || config.consumeUsage;
+
+    /**
+     * A hook event that fires before an item roll is configured.
+     * @function dnd5e.preRollConfiguration
+     * @memberof hookEvents
+     * @param {Item5e} item                      Item being rolled.
+     * @param {ItemRollConfiguration} config     Configuration data for the item roll being prepared.
+     * @param {ItemRollOptions} options          Additional options used for configuring item rolls.
+     * @returns {boolean}                        Explicitly return false to prevent item from being rolled.
+     */
+    if ( Hooks.call("dnd5e.preRoll", item, config, options) === false ) return;
```

Comment:
I think this hook name is possibly too general. The method here is effectively "play an item card to chat" - the method is called `roll` because in early versions of `dnd5e` that is typically what resulted from this workflow.... but a side comment is that perhaps `roll` is not the right name for this method either.

Regardless, I think creating hooks for `dnd5e.preRoll` or `dnd5e.roll` are too easily confused with functionality that might apply at the time that an actual dice roll occurs rather than as part of the item chat card.

Reply:
How about `dnd5e.preUseItem` & `dnd5e.useItem` (and other related hook changes) to match the term of `_getUsageUpdates`?